### PR TITLE
Implement IQ stream direct to memory

### PR DESCRIPTION
**Pull request by aromanielloNTIA**
- Implements `IQSTREAM_Acquire` which streams IQ data directly to a NumPy array in memory. This reduces IQ streaming overhead substantially (~99%, depending on IQ capture duration) and allows for longer IQ captures than supported by `IQSTREAM_Tempfile` methods
- Implements the `IQSTREAM_GetIQData` function and the `_IQStreamIQInfo` struct from the underlying API to support this change
- Corrects the constant for the maximum frequency reference user setting string length
- Fixes an exception occurring when frequency reference user setting strings have checksum integer lengths over 2
- Increments version number to 1.3.0 (changes are fully backwards compatible)
- Removes a bunch of superfluous `global` statements in class methods
- Updates documentation to reflect these changes

These changes have been tested locally using a custom script and the included unit tests on RSA306, RSA306B, and RSA507A. When measuring noise, the new `IQSTREAM_Acquire` method was shown to deliver the same RF power result (to within .01 dB using 5s of IQ data at 56 MHz sampling rate) when compared to `IQSTREAM_Tempfile`. In this case, the new method reduced streaming overhead time from 3.57 seconds to under 17 ms (>99.5% improvement).

### CODE REVIEW

File: src/rsa_api/rsa_api.py [hunk 1]
```diff
@@ -1379,6 +1385,50 @@ def IQSTREAM_GetEnable(self) -> bool:
...
+        c_iqdata = np.ctypeslib.as_ctypes(iqdata)
+        iqlen = c_int32()
+        iqinfo = _IQStreamIQInfo()
+
+        self.err_check(
+            self.rsa.IQSTREAM_GetIQData(byref(c_iqdata), byref(iqlen), byref(iqinfo))
+        )
+
+        del c_iqdata
+        return iqdata, iqlen.value, iqinfo
```
Comment (jhazentia) [inline src/rsa_api/rsa_api.py:1430]: It looks like method return type needs to be updated to include iqlen

Comment (aromanielloNTIA) [reply src/rsa_api/rsa_api.py:1430]: Good catch- fixed in [46002b1](https://github.com/NTIA/tekrsa-api-wrap/pull/13/commits/46002b1b3c5ffc54f128cf349e8c8002d14f7f98)

File: src/rsa_api/rsa_api.py [hunk 2]
```diff
@@ -2499,3 +2542,81 @@ def DEVICE_GetTemperature(self, unit: str = "celsius") -> float:
...
+            iq_samples_received += iq_block_len
+
+        self.IQSTREAM_Stop()
+        self.DEVICE_Stop()
+
+        assert len(iqdata) == iq_samples_requested
+
+        if return_status:
+            iq_status = self.IQSTREAM_StatusParser(iqinfo, not return_status)
+            return iqdata, iq_status
```
Comment (jhazentia) [inline src/rsa_api/rsa_api.py:2620]: Could optionally update method return signature to include iq_status

Comment (aromanielloNTIA) [reply src/rsa_api/rsa_api.py:2622]: Good call. I made this correction in [af8ca02](https://github.com/NTIA/tekrsa-api-wrap/pull/13/commits/af8ca026899bc2359e396ff741f921c11770574b) for `IQSTREAM_Acquire` as well as `IQSTREAM_Tempfile` and `IQSTREAM_Tempfile_NoConfig` which work the same way.

**Review by jhazentia — APPROVED** [review]
Looks good

**Pull request merged**